services/shhext/api.go

```diff
@@ -186,7 +188,42 @@ func (api *PublicAPI) GetNewFilterMessages(filterID string) ([]*whisper.Message,
 	if err != nil {
 		return nil, err
 	}
-	return api.service.deduplicator.Deduplicate(msgs), err
+
+	dedupMessages := api.service.deduplicator.Deduplicate(msgs)
+
+	// Attempt to decrypt message, otherwise leave unchanged
+	for _, msg := range dedupMessages {
+
+		// Msg.Dst is empty is a public message, nothing to do
+		if msg.Dst != nil {
+			// There's probably a better way to do this
+			keyBytes, err := hexutil.Bytes(msg.Dst).MarshalText()
```

Comment:
Is there a better way for this?

Reply:
`fmt.Sprintf()`? It should call MarshalText internally.

Reply:
thanks!